examples/simulation/Tutorial4/src/tools/FairTutorialDet4GeoHandler.cxx

```diff
@@ -102,7 +102,7 @@ Int_t FairTutorialDet4GeoHandler::VolId(const Text_t* name) const
         if (name[len] != ' ') {
             return VolIdGeo(name);
         }
-        strncpy(sname, name, len);
+        memcpy(sname, name, len);
```

Comment:
There are actually two more issues present in this code:
* buffer overflow if `name` is longer than 20 characters
* buffer underflow if `name` is empty

I suggest something like this (untested):

```diff
@@ -100,8 +100,5 @@
-        char sname[20];
-        Int_t len = strlen(name) - 1;
-        if (name[len] != ' ') {
-            return VolIdGeo(name);
-        }
-        strncpy(sname, name, len);
-        memcpy(sname, name, len);
-        sname[len] = 0;
-        return VolIdGeo(sname);
+        std::string volname(name);
+        while (!volname.empty() && volname.back() == ' ') {
+          volname.pop_back();
+        }
+        return VolIdGeo(volname.c_str());
```

This also does a proper `rtrim` which I assume was the original intention anyways (if not, replace the `while` with an `if`).

Reply:
Done